src/utils/featheryClient/index.ts

```diff
@@ -72,7 +74,50 @@ export const updateRegionApiUrls = (region: string) => {
   }
 };
 
+/**
+ * The number of milliseconds waited until another submitCustom call
+ */
+const SUBMIT_CUSTOM_DEBOUNCE_WINDOW = 1000;
+
 export default class FeatheryClient extends IntegrationClient {
+  /**
+   * Used to aggregate field value updates for sucessive calls to
+   * submitCustom within the debounce window
+   */
+  pendingSubmitCustomUpdates: { [key: string]: any };
+
+  /**
+   * Debounced implementation of submitCustom
+   */
+  debouncedSubmitCustom: DebouncedFunc<(override: boolean) => Promise<void>>;
+
+  /**
+   * If there is a pending invocation of submitCustom, this method calls it immediately
+   */
+  flushPendingSubmitCustomUpdates: (
+    override?: boolean
+  ) => Promise<void> | undefined;
+
+  constructor(
+    formKey = '',
+    ignoreNetworkErrors?: any,
+    draft = false,
+    bypassCDN = false
+  ) {
+    super(formKey, ignoreNetworkErrors, draft, bypassCDN);
+    this.pendingSubmitCustomUpdates = {};
+    this.debouncedSubmitCustom = debounce(
+      this._debouncedSubmitCustom.bind(this),
+      SUBMIT_CUSTOM_DEBOUNCE_WINDOW
+    );
+    this.flushPendingSubmitCustomUpdates = (override?: boolean) => {
```

Comment:
i'm not sure this method is necessary actually. if there is data to flush, there will be a debounced, pending submitCustom call as well so no need to call this again or have this helper function. .flush() can be called directly.

Reply:
I think wrapping the flush call in this method creates for a cleaner API long term. 

There's currently 3 instances where we flush. Since the `lodash.flush` implementation has this [caveat](https://github.com/lodash/lodash/issues/4185#issuecomment-462388355) that requires us to 
1. call the debounced function
2. then  call `flush()`

It's easier to debug and refactor the flush call if its defined in one place.

Otherwise, there's going to be 3 instances of:

```{typescript}
this.debouncedSubmitCustom(override ?? true);
this.debouncedSubmitCustom.flush();
```